### DISCUSSION TITLE
Start porting the backend REST API documentation to OpenAPI

### DISCUSSION THREAD

**Discussion by prettyirrelevant** [rotki/rotki#6682] [Ideas]
## Abstract
Currently, the API documentation is lacking in the proper structure of response specifically in a situation whereby the response is an array of objects or a nested object.
## Motivation
Porting to a new solution will improve developer experience in understanding how the API works and proper description of fields being returned.

## Specification
- One approach to tackling this is to write the API docs using Swagger and host it separately.
- Another approach entails porting the entire Rotki documentation to a different platform although API will still be written using Swagger.

**Comment by kelsos** [comment]
So we had a discussion in person about the benefits of porting the documentation to OpenAPI (Swagger) and the benefits it provides.

- Hosted version can provide tooling to quickly interact with local backend
- Can be imported to tools like Postman/Insomnia

We also had a look at trueblocks' [API documentation](https://trueblocks.io/api/), and they use [redoc](https://github.com/Redocly/redoc) for the generation of the webpage.

We need to check if there is a way to automatically generate the OpenAPI spec directly from the flask backend code.
